amivapi/events/validation.py

```diff
@@ -138,68 +145,81 @@ def _validate_email_signup_must_be_allowed(self, enabled, field, value):
-            }
-
-            for key, value in enforced_fields.items():
-                if key not in json_data or json_data[key] != value:
-                    self._error(field,
-                                "'{key}' is required to be set to '{value}'"
-                                .format(key=key, value=value))
+            value: field value
+
+        1: https://json-schema.org
 
+        The rule's arguments are validated against this schema:
+        {'type': 'boolean'}
+        """
+        if enabled:
             try:
+                json_data = json.loads(value)
+            except json.JSONDecodeError as error:
+                self._error(field,
+                            "Invalid json, parsing failed with exception: %s"
+                            % error)
+
+            else:
+                # validate if these fields are included exactly as given
+                # (we, e.g., always require objects so UI can rely on this)
+                enforced_fields = {
+                    '$schema': 'http://json-schema.org/draft-04/schema#',
+                    'type': 'object',
+                    'additionalProperties': False
+                }
+
+                for key, val in enforced_fields.items():
+                    if key not in json_data or json_data[key] != val:
+                        self._error(field,
+                                    "'%s' is required to be set to '%s'"
+                                    % (key, val))
+
                 # now check if it is entirely valid jsonschema
-                v = Draft4Validator(json_data)
+                validator = Draft4Validator(json_data)
                 # by default, jsonschema specification allows unknown properties
                 # We do not allow these.
-                v.META_SCHEMA['additionalProperties'] = False
-                v.check_schema(json_data)
-            except SchemaError as e:
-                self._error(field, "does not contain a valid schema: %s"
-                            % str(e))
+                validator.META_SCHEMA['additionalProperties'] = False
+
+                try:
+                    validator.check_schema(json_data)
+                except SchemaError as error:
+                    self._error(field, "does not contain a valid schema: %s"
+                                % error)
 
     # Eve doesn't handle time zones properly. Its always UTC but sometimes
     # the timezone is included, sometimes it isn't.
 
     def _get_time(self, fieldname):
-        """Retrieve time field from document or _original_document."""
+        """Retrieve time field from document or original document."""
         # Try to pick the value from document first, fall back to original
         time = self.document.get(fieldname)
         if time is None:
-            time = (self._original_document[fieldname]
-                    if self._original_document else None)
-
-        if time is None:
-            return None
```

Comment:
Why did you remove these two line? If time is now None here, this will crash the API as None does not have a replace function, which is still called in the next line.

Reply:
I merged it into the return statement: `return time.replace(tzinfo=None) if time is not None else None`

Do you think this looks too confusing? Then I can change it back^^